### ISSUE TITLE
Fix handle "jumping" when list items have different sizes

### ISSUE THREAD

**Issue by micHar** [FutureMind/recycler-fast-scroll#5]

**Comment by micHar** [comment]
When an item is significantly smaller or bigger than an average one in the list then the scroller jumps a little bit. No idea how to cope with this problem atm.

**Comment by micHar** [comment]
Maybe it should calculate an average size as the list scrolls by. Keep all the sizes in some sort of container and calculate average onScrolled.

**Comment by junyuecao** [comment]
I'm not sure if it's too late but I found the following solution.
In the `ScrollListener` of FastScroll.java (around line 205)
change it to this:
```
        @Override
        public void onScrolled(RecyclerView rv, int dx, int dy) {
            if (handle != null && !manuallyChangingPosition && recyclerView.getChildCount() > 0) {
                float range;
                float offset;
                if (isVertical()) {
                    range = recyclerView.computeVerticalScrollRange();
                    offset = recyclerView.computeVerticalScrollOffset();
                } else {
                    range = recyclerView.computeHorizontalScrollRange();
                    offset = recyclerView.computeHorizontalScrollOffset();
                }
                setHandlePosition(offset / range);
            }
        }
```

**Comment by qhutch** [comment]
you can fix this by changing the scroll listener (line 198 of the FastScroller.java file)

```
public void onScrolled(RecyclerView rv, int dx, int dy) {
            if(handle!=null && !manuallyChangingPosition && recyclerView.getChildCount() > 0) {
                float relativePos;
                if(isVertical()) {
                    int offset = recyclerView.computeVerticalScrollOffset();
                    int extent = recyclerView.computeVerticalScrollExtent();
                    int range = recyclerView.computeVerticalScrollRange();

                    relativePos = offset / (float)(range - extent);
                } else {
                    int offset = recyclerView.computeHorizontalScrollOffset();
                    int extent = recyclerView.computeHorizontalScrollExtent();
                    int range = recyclerView.computeHorizontalScrollRange();

                    relativePos = offset / (float)(range - extent);
                }
                //setHandlePosition(recyclerViewAbsoluteScroll / recyclerViewOversize);
                setHandlePosition(relativePos);
            }
        }
```

**Comment by TostF** [comment]
Fixed via - #12
